src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -2362,6 +2396,24 @@ public IEnumerable<TimeSeriesDeletedRangeItem> GetDeletedRangesFrom(DocumentsOpe
             }
         }
 
+        public IEnumerable<TimeSeriesDeletedRangeItem> GetDeletedRangesByBucketFrom(DocumentsOperationContext context, int bucket, long etag, long toEtag = long.MaxValue)
+        {
+            var table = new Table(DeleteRangesSchema, context.Transaction.InnerTransaction);
+
+            using (DocumentsStorage.GetBucketAndEtagByteString(context.Allocator, bucket, etag, out var buffer))
+            using (Slice.External(context.Allocator, buffer, buffer.Length, out var keySlice))
+            using (Slice.External(context.Allocator, buffer, buffer.Length - sizeof(long), out var prefix))
+            {
+                foreach (var result in table.SeekForwardFromPrefix(DeleteRangesSchema.DynamicKeyIndexes[DeletedRangesBucketAndEtagSlice], keySlice, prefix, 0))
+                {
+                    var item = CreateDeletedRangeItem(context, ref result.Result.Reader);
+                    if (item.Etag > toEtag)
```

Comment:
what scenario will we use `toEtag` ?

Reply:
I added `toEtag` to stay consistent with `GetDeletedRangesFrom(DocumentsOperationContext context, long etag, long toEtag = long.MaxValue)`
but it seems that we never pass `toEtag` to `GetDeletedRangesFrom`

removed `toEtag` from both `GetDeletedRangesFrom` and `GetDeletedRangesByBucketFrom`